File: hyoungkyu/230720/[BOJ] 1522 문자열 교환.py
# 실버1 / 40ms
S = input()*2
len_a = S.count('a') // 2

min_v = len_a
for i in range(len(S)-len_a+1):
    min_v = min(S[i:i+len_a].count('b'), min_v)
print(min_v)

'''
8 12
aabbaaabaaba
aabbaaab - 3
 abbaaaba - 3
  bbaaabaa - 3
   baaabaab - 3
    aaabaaba - 2
'''

File: hyoungkyu/230720/[BOJ] 1522 문자열 교환.java
// 실버1 / 136ms
import java.io.*;

class Solution
{
	public static void main(String args[]) throws Exception
	{
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
		// StringBuilder sb = new StringBuilder();				// 다 모아서 한번에 출력 => 이거 안하면 시간초과;;
		String S = br.readLine();
        int lenA = (int) S.chars().filter(i -> String.valueOf((char) i).equals("a")).count();
        S = S + S;
        int minV = S.length();
        for (int i=0; i<S.length()-lenA; i++) {
            int tmp = (int) S.substring(i, i+lenA).chars().filter(j -> String.valueOf((char) j).equals("b")).count();
            minV = Math.min(minV, tmp);
        }
        System.out.println(minV);
	}
}